File: packages/engine/tests/examples/ant_foraging/dependencies/@hash/move-in-direction/src/behaviors/move_in_direction.rs
use crate::prelude::{AgentState, Context, SimulationResult};

// deps: 'direction' = a vector (x, y)
// moves the agent in its current direction.
pub fn move_in_direction(
    mut state: AgentState,
    _context: &Context,
) -> SimulationResult<AgentState> {
    if let Some(dir) = &state.direction {
        let (dx, dy) = (dir.x(), dir.y());
        let pos = state.get_pos_mut()?;
        pos[0] += dx;
        pos[1] += dy;
    }

    Ok(state)
}

File: packages/engine/tests/examples/city_infection_model/src/behaviors/move_in_direction.js
/**
 * @param {AgentState} state of the agent
 * @param {AgentContext} context of the agent
 */
function behavior(state, context) {
  if (state.direction) {
    state.position = state.position.map((x, i) => x + state.direction[i]);
  }
}
